.github/scripts/validate_scale_config.py

```diff
@@ -0,0 +1,338 @@
+                )
+                errors_found = True
+                continue
+
+        if not compare_runner_types(
+            base_runner_type,
+            runner_types[base_runner_type],
+            amz2023_runner_type,
+            runner_types[amz2023_runner_type],
+            ignore_ami=True,
+        ):
+            errors_found = True
+
+    return not errors_found
+
+
+def is_consistent_across_configs(
+    source_config: Dict[str, Dict[str, str]],
+    dest_config: Dict[str, Dict[str, str]],
+    expected_prefix: str,
+) -> bool:
+    """
+    Validate that every runner type in the source_config has a corresponding runner type in the dest_config
+    where the dest_config has the expected_prefix added
+    """
+    errors_found = False
+
+    # Every entry in the source_config should be in the dest_config with
+    # the same settings, except that the runner_type should have the expected_prefix
+    for source_runner_type in source_config:
+        dest_runner_type = f"{expected_prefix}{source_runner_type}"
+
+        if dest_runner_type not in dest_config:
+            print(
+                f"Runner type {source_runner_type} does not have a corresponding {dest_runner_type} runner type"
+            )
+            errors_found = True
+            continue
+
+        if not compare_runner_types(
+            source_runner_type,
+            source_config[source_runner_type],
+            dest_runner_type,
+            dest_config[dest_runner_type],
+        ):
+            errors_found = True
+
+    return not errors_found
+
+
+def generate_repo_scale_config(
+    source_config_file: str, dest_config_file: str, expected_prefix: str
+) -> None:
+    """
+    Generate the new scale config file with the same layout as the original file,
+    but with the expected_prefix added to the runner types
+    """
+
+    print(f"Generating updated {dest_config_file}")
+    source_config = load_scale_config(source_config_file)
+    base_runner_types = set(source_config[RUNNER_TYPE_CONFIG_KEY].keys())
+
+    with open(source_config_file, "r") as f:
+        source_config_lines = f.readlines()
+
+    with open(dest_config_file, "w") as f:
+        f.write(
+            """
+# This file is generated by .github/scripts/validate_scale_config.py in test-infra
+# It defines runner types that will be provisioned by by LF Self-hosted runners
+
+"""
+        )
+        for line in source_config_lines:
+            # Any line that has a runner type should have the expected prefix added.
+            # Otherwise we can just copy the line over
+            entry = line.strip(" :\n")
+            if entry in base_runner_types:
+                # We found a runner type. Give it the expected prefix
+                line = line.replace(entry, f"{expected_prefix}{entry}")
+
+            f.write(line)
+
+
+def load_scale_config(scale_config_path: str) -> Dict[str, Any]:
```

Comment:
Imo this better describes what this function is trying to achieve, isn't it?
```suggestion
def load_json_from_file(path: str) -> Dict[str, Any]:
```

Reply:
Renamed it to something in a similar vein